azure/ConsiderationsForServiceDesign.md

```diff
@@ -84,6 +84,74 @@ As you build out your service and API, there are a number of decisions that can
 
 :ballot_box_with_check: **YOU SHOULD** implement [conditional requests](https://tools.ietf.org/html/rfc7232) early. This allows you to support concurrency, which tends to be a concern later on.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a great developer experience.
+
+Resources are described by nouns. Resource and property names must be descriptive and easy for customers to understand.
+Avoid names that refer to service implementation details, e.g. "TreeLeafNode", or simply describe the structure of value without
+describing its purpose, e.g. "KeyValuePair".
```

Comment:
It would be good to say what names should be used instead and give examples, e.g. use names that correspond to user scenarios (what the user is trying to acomplish), e.g. ConfigurationSetting instead of KeyValuePair.

Reply:
Good suggestion. I have fixed.